Framework/LiveData/src/Kafka/KafkaEventStreamDecoder.cpp

```diff
@@ -243,13 +372,71 @@ void KafkaEventStreamDecoder::eventDataFromMessage(const std::string &buffer) {
   } else {
     periodBuffer = m_localEvents[0];
   }
-  for (decltype(nEvents) i = 0; i < nEvents; ++i) {
-    auto &spectrum = periodBuffer->getSpectrum(
-        m_specToIdx[static_cast<int32_t>(detData[i])]);
-    spectrum.addEventQuickly(TofEvent(static_cast<double>(tofData[i]) *
-                                          1e-3, // nanoseconds to microseconds
-                                      pulseTime));
+
+  auto numGroups = tofData.size() / MAX_BUFFERGROUP_EVENTS;
+  std::vector<size_t> sizes(numGroups + 1, MAX_BUFFERGROUP_EVENTS);
+  sizes[numGroups] = tofData.size() % MAX_BUFFERGROUP_EVENTS;
+  size_t start = 0;
+
+  auto starttime = std::chrono::system_clock::now();
+  for (auto &grpSize : sizes) {
+    if (grpSize == 0)
+      continue;
+
+    BufferGroup buffGrp;
```

Comment:
Buffers should be reused, otherwise we pay a lot for constantly reallocating them.